homework/index.js

```diff
@@ -29,19 +29,124 @@
     });
     return elem;
   }
+  // table of repository's basic information
+  function createDetailsTable(index, data, container) {
+    const repositoryData = createAndAppend('div', container, { class: 'left-div white-frame' });
+    const table = createAndAppend('table', repositoryData);
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, {
+      text: 'Repository :',
+      class: 'label',
+    });
+    // manually create td with a link
+    const td = createAndAppend('td', tr);
+    createAndAppend('a', td, {
+      href: data[index].html_url,
+      text: data[index].name,
+      target: '_blank',
+    });
+    // a function to create td without link
+    function createTd(textKey, textContent) {
```

Comment:
You should avoid nested functions like this. The reason is that nested function obscure where the data comes from that they need. A nested function has access to the data of its outer function. In this case, the `createTd` function accesses the `tbody` variable from the scope of its parent function, but that is not immediately obvious. You can easily move this function out of the `createDetailsTable` function by passing the `tbody` variable as a third parameter, e.g.:

```js
function createTd(textKey, textContent, tbody) {
  ...
}

function createDetailsTable(index, data, container) {
  ...
  createTd('Description', data[index].description, tbody);
  ...
}
```
 Now it is absolutely clear what the `createTd` function needs from the outside world to do its work.

Reply:
OK, It works so